### PR TITLE
The PHP DOM extension is required for JUnit

### PR DESCRIPTION
**Pull request by pfrenssen**
I'm getting the following error when running a test scenario on a minimal Ubuntu Bionic image:

```
Fatal error: Class 'DOMDocument' not found (Behat\Testwork\Call\Exception\FatalThrowableError)
```

It is thrown in [`Behat\Testwork\Output\Printer\JUnitOutputPrinter`](https://github.com/Behat/Behat/blob/master/src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php#L62).

### CODE REVIEW

**Comment by jakzal** [comment]
It's only required by the junit output printer and not technically needed for other Behat features to work.

Would it make more sense if we checked if the extension is present in the JUnitOutputPrinter class, and throw an exception if it's not?

**Comment by pfrenssen** [comment]
Yes that is also an option, then we can move it to `suggest`.

File: src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php [hunk 1]
```diff
@@ -57,6 +57,10 @@ public function __construct(FilesystemOutputFactory $outputFactory)
      */
     public function createNewFile($name, array $testsuitesAttributes = array())
     {
+        // This requires the DOM extension to be enabled.
+        if (!extension_loaded('dom')) {
+            throw new \Exception('The PHP DOM extension is required to generate JUnit reports.');
```
Comment (jakzal) [inline src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php]: I'd throw something more specific, like `RuntimeException`.

Comment (jakzal) [reply src/Behat/Testwork/Output/Printer/JUnitOutputPrinter.php]: There are also some more Printer specific exceptions in this package: https://github.com/Behat/Behat/tree/master/src/Behat/Testwork/Output/Exception

I think the intention might've been for Printer implementations to throw exceptions implementing the PrinterException.

**Comment by pfrenssen** [comment]
Sorry for the delay, this fell off my radar after I moved on to other tasks. I have updated the PR with a dedicated exception which implements `PrinterException`.

**Pull request merged**